src/extension.ts

```diff
@@ -551,37 +565,79 @@ function configureExtension(context: vscode.ExtensionContext) {
-      updateExploreDecorations();
-    }
-    else if (event.affectsConfiguration("vectorcastTestExplorer.verboseLogging")) {
-      adjustVerboseSetting();
-    }
-    else if (event.affectsConfiguration("vectorcastTestExplorer.configurationLocation")){
-      updateConfigurationOption (event);
+    // post configuration, we handle changes to all options ...
+    if (alreadyConfigured) {
+      // This function gets triggered when any option at any level (user, workspace, etc.)
+      // gets changed.  The event parameter does not indicate what level has been
+      // edited but you can use the 
+
+      if (event.affectsConfiguration("vectorcastTestExplorer.decorateExplorer")) {
+        updateExploreDecorations();
+      }
+      else if (event.affectsConfiguration("vectorcastTestExplorer.verboseLogging")) {
+        adjustVerboseSetting();
+      }
+      else if (event.affectsConfiguration("vectorcastTestExplorer.configurationLocation")){
+        updateConfigurationOption (event);
+      }
+      else if (
+        event.affectsConfiguration(
+          "vectorcastTestExplorer.vectorcastInstallationLocation"
+        )
+      ) {
+        // if the user changes the path to vcast, we need to reset the values
+        // for clicast and vpython path etc.
+        if (checkIfInstallationIsOK()) {
+          resetCoverageData();
+          buildTestPaneContents();
+          updateCOVdecorations();
+        }
+      }
     }
-    else if (
-      event.affectsConfiguration(
-        "vectorcastTestExplorer.vectorcastInstallationLocation"
-      )
-    ) {
-      // if the user changes the path to vcast, we need to reset the values
-      // for clicast and vpython path etc.
-      if (checkIfInstallationIsOK()) {
-        resetCoverageData();
-        buildTestPaneContents();
-        updateCOVdecorations();
+    // pre-configuration, we only handle changes to the vcast installation location
+    else {
+      if (event.affectsConfiguration("vectorcastTestExplorer.vectorcastInstallationLocation")) {
+        // this call will check if the new value is valid, 
+        // and if so, perform extension activation
+        checkPrerequisites (context)
       }
     }
   });
+
+  // Command: vectorcastTestExplorer.newEnviroVCAST ////////////////////////////////////////////////////////
+  let newEnviroVCASTCommand = vscode.commands.registerCommand(
+    "vectorcastTestExplorer.newEnviroVCAST",
+    (args: Uri, argList: Uri[]) => {
+
+      // contains a check for already configured, so no work will be done in that case
+      checkPrerequisites(context)
+      if (alreadyConfigured) {
+        // arg is the actual item that the right click happened on, argList is the list
+        // of all items if this is a multi-select.  Since argList is always valid, even for a single
+        // selection, we just use this here.
+        if (argList) {
+          newEnvironment(argList);
+        }
```

Comment:
@johnpaliotta Are we sure we kept the desired behavior when `alreadyConfigured` is still false when making the new environment? It seems correct to me, since we only had `if (argList) {newEnvironment(argList);}` in 
`"vectorcastTestExplorer.newEnviroVCAST"` before and things were being triggered properly, but I think it'd make sense to double check

Reply:
Yes, this is right because check prerequisites will either "pass" and set the already configured, or will fail and open the settings and output pane so you can fix things.